tests/test_dask_image/test_ndinterp/test_affine_transformation.py

```diff
@@ -0,0 +1,314 @@
+                         [(15, 10)])
+def test_affine_transform_modes(n,
+                                interp_mode,
+                                input_output_shape_per_dim,
+                                input_output_chunksize_per_dim,
+                                helpers):
+
+    kwargs = dict()
+    kwargs['n'] = n
+    kwargs['interp_mode'] = interp_mode
+    kwargs['input_output_shape_per_dim'] = input_output_shape_per_dim
+    kwargs['input_output_chunksize_per_dim'] = input_output_chunksize_per_dim
+    kwargs['interp_order'] = 0
+
+    helpers.test_affine_transform(**kwargs)
+
+
+@pytest.mark.parametrize("interp_mode",
+                         ['wrap', 'reflect', 'mirror'])
+def test_affine_transform_unsupported_modes(interp_mode, helpers):
+
+    kwargs = dict()
+    kwargs['interp_mode'] = interp_mode
+
+    with pytest.raises(NotImplementedError):
+        helpers.test_affine_transform(**kwargs)
+
+
+def test_affine_transform_numpy_input():
+
+    image = np.ones((3, 3))
+    image_t = da_ndinterp.affine_transform(image, np.eye(2), [0, 0])
+
+    assert image_t.shape == image.shape
+    assert (image == image_t).min()
+
+
+def test_affine_transform_type_consistency():
+
+    image = da.ones((3, 3))
+    image_t = da_ndinterp.affine_transform(image, np.eye(2), [0, 0])
+
+    assert isinstance(image, type(image_t))
+    assert isinstance(image[0, 0].compute(), type(image_t[0, 0].compute()))
+
+
+def test_affine_transform_type_consistency_gpu():
+
+    pytest.importorskip("cupy", minversion="6.0.0")
+
+    image = da.ones((3, 3))
+    image_t = da_ndinterp.affine_transform(image, np.eye(2), [0, 0])
+
+    import cupy as cp
+    image.map_blocks(cp.asarray)
+
+    assert isinstance(image, type(image_t))
+    assert isinstance(image[0, 0].compute(), type(image_t[0, 0].compute()))
+
+
+def test_affine_transform_no_output_shape_or_chunks_specified():
+
+    image = da.ones((3, 3))
+    image_t = da_ndinterp.affine_transform(image, np.eye(2), [0, 0])
+
+    assert image_t.shape == image.shape
+    assert image_t.chunks == tuple([(s,) for s in image.shape])
+
+
+def test_affine_transform_prefilter_warning():
+
+    with pytest.warns(UserWarning):
+        da_ndinterp.affine_transform(da.ones(3), [1], [0],
+                                     order=3, prefilter=True)
+
+
+def test_affine_transform_large_input_small_output_cpu():
+    """
+    Make sure input array does not need to be computed entirely
+    """
+
+    # fully computed, this array would occupy 8TB
+    image = da.random.random([10000] * 3, chunks=(200, 200, 200))
+    image_t = da_ndinterp.affine_transform(image, np.eye(3), [0, 0, 0],
+                                           output_chunks=[1, 1, 1],
+                                           output_shape=[1, 1, 1])
+
+    # if more than the needed chunks should be computed,
+    # this would take long and eventually raise a MemoryError
+    # use pytest-timeout?
+    image_t[0, 0, 0].compute()
```

Comment:
Yes, pytest-timeout sounds like a good idea here

Reply:
Great. The test `test_affine_transform_large_input_small_output_cpu` takes around 2-3s on my macbook, so I set the timeout at 15s for now to be on the safe side.

And added a `pytest-timeout` requirement in `setup.py`.

```
test_requirements = [
    "flake8 >=3.4.1",
    "pytest >=3.0.5",
    "pytest-flake8 >=0.8.1",
    "pytest-timeout >=1.0.0",
    "tifffile >=2018.10.18",
]
```